### PR TITLE
Move build tests from GA to Gitlab

### PR DESCRIPTION
**Pull request by alalazo**
Modifications:
- [x] Remove the "build tests" workflow from GitHub Actions
- [x] Setup a similar e2e test on Gitlab

In this way we'll reduce load on GitHub Actions workflows and for e2e tests we'll benefit from the buildcache reuse granted by pipelines.

### CODE REVIEW

**Comment by alalazo** [comment]
@zackgalbreath @scottwittenburg I asked for your reviews since I may need some guidance while trying to move build tests from Github to Gitlab. Thanks a lot in advance 🙂

File: share/spack/gitlab/cloud_pipelines/stacks/build_systems/spack.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      - ruby-rake  # RubyPackage
+    - arch:
+      - '%gcc@7.5.0 arch=linux-ubuntu18.04-x86_64'
+
+  specs:
+    - matrix:
+        - - $default_specs
+        - - $arch
+
+  mirrors: { "mirror": "s3://spack-binaries-develop/build_systems" }
```
Comment (alalazo) [inline share/spack/gitlab/cloud_pipelines/stacks/build_systems/spack.yaml:29]: Do we have to do anything to create this new mirror?

Comment (scottwittenburg) [reply share/spack/gitlab/cloud_pipelines/stacks/build_systems/spack.yaml:29]: No, nothing required as long as the bucket (in this case `spack-binaries-develop`) already exists.  You should be all set.

**Review by scottwittenburg — APPROVED** [review]
All looks good to me, and the pipeline passed!  👍

**Comment by alalazo** [comment]
@tgamblin Merging this PR, since it got approved Let me know if there are further changes, I'll do them in a subsequent PR.

**Comment by tgamblin** [comment]
LGTM!  thanks @alalazo and @scottwittenburg!

**Pull request merged**